addons/event/static/src/template_reference_field/field_event_mail_template_reference.js

```diff
@@ -0,0 +1,28 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { EventMailTemplateMany2OneField } from "./field_event_mail_template_many2one";
+import { ReferenceField, referenceField } from "@web/views/fields/reference/reference_field";
+
+export class EventMailTemplateReferenceField extends ReferenceField {
```

Comment:
So the goal of this whole thing is to display that small enveloppe here right:
![image](https://github.com/odoo/odoo/assets/41282536/7d711965-5008-4238-8864-f80c3d0b4d9e)

You probably already considered this, but would it be possible to achieve the same result with a lower footprint?
Some random ideas:
- Change the template display_name to include an icon (emoji?) before its name (if we don't want this always applied, could be done via a specific context key)
- Add your "notification_type" computed column before the reference field, with some CSS to make it hidden when in edit mode?

EDIT: considering we have lots of different target models, option 1 might not be doable

Reply:
The irony of a REF PR adding a field just to display icons isn't lost on me ^^ 

> regular text emotions

Then it's user-font-dependent which isn't the best.
Looks a bit out of place to me
![image](https://github.com/odoo/odoo/assets/43653136/6bd099ea-aefc-453b-a6f4-0463c8f6ccae)

If that wasn't a limiting factor indeed we could just have icons in the selection of the Reference field and just always display it, or something like that. 

Realistically it's 10 lines of actual non-boiler-plate code to do this so I don't think we really need to look for a hack just to avoid adding a field.